openmeetings-core/src/main/java/org/apache/openmeetings/core/remote/StreamProcessor.java

```diff
@@ -431,6 +433,10 @@ public void stopRecording(Client c) {
 			return;
 		}
 		kHandler.getRoom(c.getRoomId()).stopRecording(this, c);
+		Optional<StreamDesc> osd = c.getScreenStream();
```

Comment:
In face as seen by the Destroy events the stopRecord actually stops the MediaStream. So stopping the Stream makes sense.
Also this solves the problem to not have any KStream ghost connections AND you can restart the recording.

Reply:
I would write it using modern syntax:
```
		c.getScreenStream().ifPresent(sd -> {
			if (!sd.hasActivity(Activity.SCREEN)) {
				pauseSharing(c, sd.getUid());
			}
		});
```